boards/riscv/tlsr9518adk80d/pin_func.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2021 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __TELINK_TLSR9518ADK80D_PIN_FUNC_H_
+#define __TELINK_TLSR9518ADK80D_PIN_FUNC_H_
+
+#define BIT(n)                 (1<<(n))
+
+#define PINMUX_FUNC_A          0
+#define PINMUX_FUNC_B          1
+#define PINMUX_FUNC_C          2
+
+/**
+ *  @brief  Define GPIO types
+ */
+#define GPIO_GROUPA            0x000
+#define GPIO_GROUPB            0x100
+#define GPIO_GROUPC            0x200
+#define GPIO_GROUPD            0x300
+#define GPIO_GROUPE            0x400
+
+#define GPIO_PA0               (GPIO_GROUPA | BIT(0))
+#define GPIO_PA1               (GPIO_GROUPA | BIT(1))
+#define GPIO_PA2               (GPIO_GROUPA | BIT(2))
+#define GPIO_PA3               (GPIO_GROUPA | BIT(3))
+#define GPIO_PA4               (GPIO_GROUPA | BIT(4))
+#define GPIO_PA5               (GPIO_GROUPA | BIT(5))
+#define GPIO_PA6               (GPIO_GROUPA | BIT(6))
+#define GPIO_PA7               (GPIO_GROUPA | BIT(7))
+
+#define GPIO_PB0               (GPIO_GROUPB | BIT(0))
+#define GPIO_PB1               (GPIO_GROUPB | BIT(1))
+#define GPIO_PB2               (GPIO_GROUPB | BIT(2))
+#define GPIO_PB3               (GPIO_GROUPB | BIT(3))
+#define GPIO_PB4               (GPIO_GROUPB | BIT(4))
+#define GPIO_PB5               (GPIO_GROUPB | BIT(5))
+#define GPIO_PB6               (GPIO_GROUPB | BIT(6))
+#define GPIO_PB7               (GPIO_GROUPB | BIT(7))
+
+#define GPIO_PC0               (GPIO_GROUPC | BIT(0))
+#define GPIO_PC1               (GPIO_GROUPC | BIT(1))
+#define GPIO_PC2               (GPIO_GROUPC | BIT(2))
+#define GPIO_PC3               (GPIO_GROUPC | BIT(3))
+#define GPIO_PC4               (GPIO_GROUPC | BIT(4))
+#define GPIO_PC5               (GPIO_GROUPC | BIT(5))
+#define GPIO_PC6               (GPIO_GROUPC | BIT(6))
+#define GPIO_PC7               (GPIO_GROUPC | BIT(7))
+
+#define GPIO_PD0               (GPIO_GROUPD | BIT(0))
+#define GPIO_PD1               (GPIO_GROUPD | BIT(1))
+#define GPIO_PD2               (GPIO_GROUPD | BIT(2))
+#define GPIO_PD3               (GPIO_GROUPD | BIT(3))
+#define GPIO_PD4               (GPIO_GROUPD | BIT(4))
+#define GPIO_PD5               (GPIO_GROUPD | BIT(5))
+#define GPIO_PD6               (GPIO_GROUPD | BIT(6))
+#define GPIO_PD7               (GPIO_GROUPD | BIT(7))
+
+#define GPIO_PE0               (GPIO_GROUPE | BIT(0))
+#define GPIO_PE1               (GPIO_GROUPE | BIT(1))
+#define GPIO_PE2               (GPIO_GROUPE | BIT(2))
+#define GPIO_PE3               (GPIO_GROUPE | BIT(3))
+#define GPIO_PE4               (GPIO_GROUPE | BIT(4))
+#define GPIO_PE5               (GPIO_GROUPE | BIT(5))
+#define GPIO_PE6               (GPIO_GROUPE | BIT(6))
+#define GPIO_PE7               (GPIO_GROUPE | BIT(7))
+
+
+/**
+ *  @brief  Define UART0 pins : TX(PA3 PB2 PD2), RX(PA4 PB3 PD3)
+ */
+#define UART0_TX_PA3            GPIO_PA3
```

Comment:
We should get these settings defined in devicetree.  See how STM32 or ATMEL SoCs are doing this.

Reply:
Done. Implemented the same solution as STM32 has.
Now it is much better.. thanks.